core/common/src/ITwinError.ts

```diff
@@ -0,0 +1,92 @@
+import { BentleyError, LoggingMetaData } from "@itwin/core-bentley";
+import { LockState } from "./IModelError";
+
+/**
+ * Detailed information about a particular object Lock that is causing the Lock update conflict.
+ * An example of a lock update conflict would be attempting to use [LockControl.acquireLocks]($backend) on an object that is already locked by another Briefcase.
+ * @beta
+ */
+export interface InUseLock {
+/** Id of the object that is causing conflict. */
+  objectId: string;
+  /**
+   * The level of conflicting lock. Possible values are {@link LockState.Shared}, {@link LockState.Exclusive}.
+   * See {@link LockState}.
+   */
+  state: LockState;
+  /** An array of Briefcase ids that hold this lock. */
+  briefcaseIds: number[];
+}
+
+/**
+ * An interface used to describe an error. This error interface should be extended when needing to throw errors with extra properties defined on them. See [[InUseLocksError]] for an example.
+ * Also see [[ITwinError.throwInUseLocksError]] and [[ITwinError.isInUseLocksError]] for examples of how to throw and check that an error is of type InUseLocksError.
+ * @beta
+ */
+export interface ITwinError {
+  /** namespace for the error */
+  namespace: string;
+  /** unique key for error, within namespace */
+  errorKey: string;
+  /** explanation of what went wrong. */
+  message: string;
+  /** stack trace of the error. */
+  stack?: string;
+  /** metadata about the exception. */
+  metadata?: LoggingMetaData;
+}
+
+/**
+ * An error raised when there is a lock conflict detected.
+ * Typically this error would be thrown by [LockControl.acquireLocks]($backend) when you are requesting a lock on an element that is already held by another briefcase.
+ * @beta
+*/
+export interface InUseLocksError extends ITwinError {
+  namespace: "IModelAccess";
+  errorKey: "InUseLocks";
+  inUseLocks: InUseLock[];
+}
+
+interface OtherError extends ITwinError {
```

Comment:
What is the purpose of this?

Reply:
I removed it, wanted to show how adding new Errors would look. I added to the comment above itwinerror that people should add a typeguard and a function to throw the error to the ITwinError namespace when extending ITwinError.

Reply:
Adding new functions to `ITwinError` namespace isn't going to scale well and won't work for errors that use types unavailable in core-common.
Did you consider either making your `InUseLocksError`-related functions free functions or putting them inside the `InUseLocksError` namespace?

Reply:
I thought adding a new namespace for each new error might be a lot but I guess it doesn't seem that bad. Part of me wanted to fill the ITwinError namespace up when possible (because otherwise it feels quite empty), but then it seems confusing for others trying to extend ITwinError when they dont fit the mould of adding to the ITwinError namespace. Switching to something that anyone could use anywhere does seem better. I switched to use namespace and suggested others do that or create standalone functions for those.